torch/csrc/jit/tensorexpr/expr.cpp

```diff
@@ -50,6 +50,36 @@ ExprHandle ExprHandle::operator<=(const ExprHandle& other) const {
   return CompareSelect::make(*this, other, CompareSelectOperation::kLE);
 }
 
+ExprHandle ExprHandle::operator&&(const ExprHandle& other) const {
+  auto compare = dynamic_cast<const CompareSelect*>(this->node());
+  CHECK(compare);
+  if (!compare->is_ret_default()) {
+    throw std::runtime_error("invalid && operator");
+  }
+  CompareSelect* ret = new CompareSelect(
+      compare->lhs(),
+      compare->rhs(),
+      other.node(),
+      new IntImm(0),
+      compare->compare_select_op());
```

Comment:
Thanks for the pointer @ZolotukhinM ! We probably need to add one comment for `CompareSelect` stating that the true expressions and false expressions are evaluated simultaneously. I was not aware of it and tried to avoid introducing the extra NOP needed in `IfThenElse`. 
Will switch to `IfThenElse`.